code/game/machinery/machinery.dm

```diff
@@ -94,32 +106,71 @@
 /obj/machinery/process() // If you dont use process or power why are you here
 	return PROCESS_KILL
 
-/obj/machinery/emp_act(severity)
-	if(use_power && !stat)
-		use_power(7500/severity)
-		. = TRUE
-	..()
+////POWER RELATED PROCS
+
+// returns true if the area has power on given channel (or doesn't require power).
+// defaults to power_channel
+/obj/machinery/proc/has_power(channel = power_channel) // defaults to power_channel
+	if(!requires_power)
+		return TRUE
+	if(!machine_powernet)
+		return FALSE
+	return machine_powernet.has_power(channel)	// return power status of the area
+
+// use active power from the local powernet
+/obj/machinery/proc/use_power(channel, amount)
+	if(!has_power())
+		return FALSE
+	if(!channel)
+		channel = power_channel
+	return machine_powernet.use_active_power(channel, amount)
+
+/obj/machinery/proc/add_static_power(channel, amount)
+	machine_powernet.adjust_static_power(channel, amount)
+
+/obj/machinery/proc/remove_static_power(channel, amount)
+	machine_powernet.adjust_static_power(channel, -amount)
+
+/*
+	* # power_change()
+	*
+	* Checks to see if the machines set power channel is powered and updates stat accordingly
+	* returns TRUE if machine's stat changes, returns FALSE if it does not, this is to make sure machines dont
+	* update their icon/overlays/lighting uneccesarily if it's contigent on NOPOWER
+	*
+	* NOTE:Subtypes of machinery should call parent here unless they change this proc's behaviour regarding NOPOWER
+*/
+/obj/machinery/proc/power_change()
+	var/old_stat = stat
+	if(has_power(power_channel) || !requires_power) //if we don't require power, we don't give a shit about the power channel!
+		stat &= ~NOPOWER
+	else
+		stat |= NOPOWER
+	return old_stat != stat //performance saving for machines that use power_change() to update icons!
+
+/// Helper proc to change the machines power usage mode, automatically adjusts static power usage to maintain perfect parity
+/obj/machinery/proc/change_power_mode(use_type = IDLE_POWER_USE)
+	if(!use_type || use_type == power_state || !machine_powernet || !power_channel) //if there is no powernet/channel, just end it here
+		return
+	switch(power_state)
+		if(IDLE_POWER_USE)
+			remove_static_power(power_channel, idle_power_consumption)
+		if(ACTIVE_POWER_USE)
+			remove_static_power(power_channel, active_power_consumption)
+
+	switch(use_type)
+		if(IDLE_POWER_USE)
+			add_static_power(power_channel, idle_power_consumption)
+		if(ACTIVE_POWER_USE)
+			add_static_power(power_channel, active_power_consumption)
```

Comment:
Yes, there's probably a more elegant way to do this. Since static power is only tracked on the powernet, if I need to alter the static power this machine uses from `idle` to `active` or to `0` power consumption then I need to take into account how static power it is using now and then add/subtract the difference, in this case I figured it was just easier to remove the static usage entirely first and then add the new one.